### DISCUSSION TITLE
Why does my client allow for a null input value

### DISCUSSION THREAD

**Discussion by Doradas0** [trpc/trpc#4005] [Q&A] [answered]
I have a procedure that takes an input that is validated using zod 
```
createItem: t.procedure
  .input(
    z.number(),
  )
  .mutation(({input}) => {
    ...
  })
```
Then when I use this in my vite client
```
  const mutation = trpc.createItem.useMutation();

  const handleSubmit = () => {
    mutation.mutate(12345);
  };
```
It works as expected except that I can pass no parameters like this with no type errors
```
  const handleSubmit = () => {
    mutation.mutate();
  };
```
For some reason the param type of this mutation is `number | void`. This then errors at runtime as expected but it would be nice to know why TS allows a null input in the first place and what I can do to make this more strict and require a number input

For context I am using these package versions
    "zod": "^3.20.6",
    "typescript": "^4.9.3",
    "vite": "^4.1.0",
    "@tanstack/react-query": "^4.24.10",
    "@trpc/client": "^10.12.0",
    "@trpc/react-query": "^10.12.0",
    "@trpc/server": "^10.12.0"

Any help or ideas would be appreciated

**Comment by sachinraja** [comment, marked as answer]
Make sure you have `"strict": true` in your tsconfig.json:
https://trpc.io/docs/react#2-enable-strict-mode

**Reply by Doradas0** [reply]
Well I feel dumb, When I split the client and server code into separate dirs I forgot move the tsconfig 🤦
Thanks for the help.
